service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -236,9 +236,23 @@ public void unadoptSite(JWTData userData, int siteId) {
       throw new WrongAdoptionStatusException(false);
     }
 
-    db.deleteFrom(ADOPTED_SITES)
-        .where(ADOPTED_SITES.USER_ID.eq(userData.getUserId()))
-        .and(ADOPTED_SITES.SITE_ID.eq(siteId))
+    db.transaction(
+        configuration -> {
+          db.deleteFrom(ADOPTED_SITES)
+              .where(ADOPTED_SITES.USER_ID.eq(userData.getUserId()))
+              .and(ADOPTED_SITES.SITE_ID.eq(siteId))
+              .execute();
+
+          this.resetTreeName(siteId);
+        });
+  }
+
+  private void resetTreeName(int siteId) {
+    int latestSiteEntryId = this.latestSiteEntry(siteId).getId();
```

Comment:
⛏️ Connecting with my other comment - in case there is no entry for the given site, can we break out of the function early so we don't call `getId()` on potentially nothing?